addons/udes_stock/models/stock_move.py

```diff
@@ -310,10 +310,20 @@ def _action_assign(self):
         don't return any extra moves that may have been created
         by refactoring.
         """
+        Move = self.env["stock.move"]
+
         res = super(StockMove, self)._action_assign()
 
         assign_moves = self.exists()._action_refactor(stage="assign")
 
+        # Unreserve any partially reserved lines if not allowed by the picking type
+        moves_to_unreserve = Move.browse()
+        partial_moves = assign_moves.filtered(lambda m: m.state == "partially_available")
```

Comment:
Action refactor might split up partially available moves into fully available and not available, so it might miss the partially available state. Do we want to do this before refactoring?